app/src/renderer/components/staking/ModalStake.vue

```diff
@@ -62,21 +70,34 @@ export default {
     TmFormMsg
   },
   data: () => ({
-    amount: 1,
-    fromIndex: 0
+    amount: 0,
+    selectedIndex: 0
   }),
+  validations() {
+    return {
+      amount: {
+        required,
+        integer: value => Number.isInteger(value),
+        between: between(1, this.fromOptions[this.selectedIndex].maximum)
+      }
+    }
+  },
   methods: {
     close() {
       this.$emit(`update:showModalStake`, false)
     },
     onStake() {
-      this.$emit(`submitDelegation`, {
-        amount: this.amount,
-        from: this.fromOptions[this.fromIndex]
-      })
-
-      this.close()
+      if (!this.$v.$invalid) {
```

Comment:
should we show some message in the case of bad validation?

Reply:
technically no since we are already disabling the button to (re)delegate in case it's invalid

Reply:
why do we need this check then?

Reply:
the check was from before and I didn't consider deleting it, do you want me to do it ?

Reply:
Done. Changed back to original